### DISCUSSION TITLE
Where did the API reference go?

### DISCUSSION THREAD

**Discussion by atomfried** [realm/realm-js#6107] [Q&A]
Hello,
under https://www.mongodb.com/docs/realm-sdks/js/latest/ there used to be an API reference for JS Realm. Now there is some auto-generated mess in which i cannot find e.g. the central realm class which exposes the functions .objects and .objectForPrimaryKey. When I click "Collection" there is no mention of the implemented JS array functions. I cannot find documentation for older Realm versions.
Maybe I am just stupid. But in my opinion an API reference should be barrier-free to the extend that even stupid people can find the most basic methods...????

**Comment by nirinchev** [comment]
Hey, with realm-js 12 we've migrated to typedoc for generating the API docs from the typescript definitions as that reduces the maintenance burden (previously we had to maintain separate api docs and typescript definition files). While no tool is perfect, we continuously strive to improve the api docs and hope to bring them to parity with the old docs in the near future. For example, the just-released 12.1.0 API docs do include the `Realm` class: https://www.mongodb.com/docs/realm-sdks/js/latest/classes/Realm-1.html.

Hope this clarifies things and if you do encounter other omissions or if you have suggestions for how to improve things, feel free to report them.

**Reply by atomfried** [reply]
Thanks, the Realm class was what I was currently looking for.
`Hope this clarifies things and if you do encounter other omissions or if you have suggestions for how to improve things, feel free to report them.`
I suggest putting the sidebar ("On This Page") in alphabetical order since the page content is in alphabetical order.
I also noticed that the Results class lists numerous js array methods while the Collection class does not. It did however in the old documentation -- I'm not quite sure which one is correct.
